webhooks/clusterclass.go

```diff
@@ -161,23 +160,25 @@ func (webhook *ClusterClass) validateCompatibleSpecChanges(old, in *clusterv1.Cl
 	allErrs = append(allErrs, webhook.validateMachineDeploymentsCompatibleChanges(old, in)...)
 
 	// Validate InfrastructureClusterTemplate changes in a compatible way.
-	allErrs = append(allErrs, webhook.validateTemplatesAreCompatible(in.Spec.Infrastructure,
-		old.Spec.Infrastructure,
-		field.NewPath("spec", "infrastructure"),
-	)...)
+	if err := check.TemplatesAreCompatible(in.Spec.Infrastructure, old.Spec.Infrastructure); err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "infrastructure"),
+			in.Spec.Infrastructure, err.Error()),
+		)
+	}
 
 	// Validate control plane changes in a compatible way.
-	allErrs = append(allErrs, webhook.validateTemplatesAreCompatible(in.Spec.ControlPlane.LocalObjectTemplate,
-		old.Spec.ControlPlane.LocalObjectTemplate,
-		field.NewPath("spec", "controlPlane"),
-	)...)
+	if err := check.TemplatesAreCompatible(in.Spec.ControlPlane.LocalObjectTemplate, old.Spec.ControlPlane.LocalObjectTemplate); err != nil {
```

Comment:
Can you please flip the order in which the templates are passed in here? (the func expects `current, desired`)

(applies to all `TemplatesAreCompatible` calls)

Otherwise the error message is wrong